pkg/api/v1/types.go

```diff
@@ -2026,16 +2026,16 @@ type NodeSelector struct {
 // A null or empty node selector term matches no objects.
 type NodeSelectorTerm struct {
 	//Required. A list of node selector requirements. The requirements are ANDed.
-	MatchExpressions []NodeSelectorRequirement `json:"matchExpressions" protobuf:"bytes,1,rep,name=matchExpressions"`
+	// +patchMergeKey=key
+	// +patchStrategy=merge
+	MatchExpressions []NodeSelectorRequirement `json:"matchExpressions" patchStrategy:"merge" patchMergeKey:"key" protobuf:"bytes,1,rep,name=matchExpressions"`
```

Comment:
I agree. Deciding whether and how to actually set those fields to use a merge strategy can be decided separately.